src/shogun/regression/KernelRidgeRegression.cpp

```diff
@@ -43,82 +42,44 @@ void CKernelRidgeRegression::init()
 	SG_ADD(&m_tau, "tau", "Regularization parameter", MS_AVAILABLE);
 }
 
-bool CKernelRidgeRegression::train_machine_pinv()
+bool CKernelRidgeRegression::solve_krr_system()
 {
-	// Get kernel matrix
-	SGMatrix<float64_t> kernel_matrix=kernel->get_kernel_matrix<float64_t>();
-	int32_t n = kernel_matrix.num_cols;
-	int32_t m = kernel_matrix.num_rows;
-	ASSERT(kernel_matrix.matrix && m>0 && n>0)
-
-	for(int32_t i=0; i < n; i++)
-		kernel_matrix.matrix[i+i*n]+=m_tau;
-
-	/* re-set alphas of kernel machine */
-	m_alpha=((CRegressionLabels*) m_labels)->get_labels_copy();
-
-	/* tell kernel machine that all alphas are needed as'support vectors' */
-	m_svs=SGVector<index_t>(m_alpha.vlen);
-	m_svs.range_fill();
-
-	if (get_alphas().vlen!=n)
+	SGMatrix<float64_t> kernel_matrix(kernel->get_kernel_matrix());
+	if (kernel_matrix.num_rows!=kernel_matrix.num_cols)
+	{
+		SG_ERROR("Kernel Matrix should be a square matrix.\n");
+	}
+	int32_t n = kernel_matrix.num_rows;
+	SGVector<float64_t> y = ((CRegressionLabels*)m_labels)->get_labels_copy();
+	if (y.vlen!=n)
 	{
 		SG_ERROR("Number of labels does not match number of kernel"
-				" columns (num_labels=%d cols=%d\n", m_alpha.vlen, n);
+			" columns (num_labels=%d cols=%d\n", m_alpha.vlen, n);
```

Comment:
these checks should be done in the constructor/setter/ or even better in the base class. This method should assume that all is set up (you should put these assumptions in the doxygen docs of it)